config/checkstyle-non-main-files-suppressions.xml

```diff
@@ -170,4 +170,7 @@
   <suppress id="properCurlCommand" files=".ci[\\/]creation-of-issue-in-other-repos\.sh"/>
   <suppress id="properCurlCommand" files="release\.sh"/>
 
+  <suppress id="kebabCaseFileNamingInConfigFolder"
+            files="config[\\/]StarterRuleSet-AllRulesByCategory\.groovy\.txt"/>
```

Comment:
Missing a reason for suppression.

Reply:
It is what I asked above. Should we suppress it at all?

Reply:
I do not see why it can't be renamed.
https://github.com/checkstyle/checkstyle/blob/master/.ci/codenarc.groovy#L10
It is a file of our creation we pass in by specific name.

@nrmancuso @Vyom-Yadav Any complaints against renaming this file to follow the new rules?

Reply:
This name is bad, please rename it. If extension doesn’t matter, I propose codenarc-rules.groovy, since I think this file is actually in groovy syntax if I remember correctly.

Reply:
I ok with proposed name. Existing name is name of file that we copied from from originally

Reply:
@nrmancuso Just `codenarc-rules.groovy`, without the `.txt` at the end? Is there a reason for this `.txt` by the way?

Reply:
> Just codenarc-rules.groovy, without the .txt at the end? 

Just `codenarc-rules.groovy` will work, but `codenarc-rules.groovy.txt` is better as IntelliJ won't run inspections on `.groovy.txt`. On `.groovy`, the complete file is red (full of errors). 

> Is there a reason for this .txt by the way?

Not really, it is stored as `.groovy.txt` in official codenarc repo, but using just `.groovy` will also work fine.

Reply:
Done. Renamed.